UnityProject/Assets/Scripts/UI/Systems/AdminTools/AdminGiveItem.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using AdminTools;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace UI.Systems.AdminTools
+{
+	public class AdminGiveItem : MonoBehaviour
+	{
+		private SpawnerSearch spawnerSearch;
+		public ConnectedPlayer selectedPlayer;
+
+		[SerializeField] private List<GameObject> recommendedGameObjects;
+
+		[SerializeField] private InputField messageInput;
+		[SerializeField] private InputField countInput;
+		[SerializeField] private TMP_InputField searchBox;
+		[SerializeField] private GameObject normalList;
+		[SerializeField] private GameObject recommendedList;
+		[SerializeField] private GameObject itemEntry;
+		[SerializeField] private GUI_AdminTools adminTools;
+
+		private void Start()
+		{
+			if (recommendedGameObjects.Count > 0)
+			{
+				spawnerSearch = SpawnerSearch.ForPrefabs(recommendedGameObjects);
+				Search(recommendedList);
+			}
+			spawnerSearch = SpawnerSearch.ForPrefabs(Spawn.SpawnablePrefabs());
+			Search(normalList);
+		}
+
+		private void OnDisable()
+		{
+			UIManager.IsInputFocus = false;
+			UIManager.PreventChatInput = false;
+		}
+
+		private void OnEnable()
+		{
+			UIManager.IsInputFocus = true;
+			UIManager.PreventChatInput = true;
+		}
+
+		public void OnSearchBoxChanged()
+		{
+			Search(normalList);
+		}
+
+		public void Search(GameObject list)
+		{
+			if (list == recommendedList)
```

Comment:
are you trying to see if both list have the same elements? Because I think this won't work in that case

Reply:
no thats not what im doing. What im doing is im seeing if i want to look for items for the Recommended area or the normal "all items" area

Reply:
just realized the thing is a gameObject as well. So the consumer code needs to have a reference to the gameObject that has the either recommended or normal list? It makes noise in my head, but maybe there is something I'm not seeing

Reply:
basically you are telling the search function which list you want to fill since there's two of them.